pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginsDirectoryVisitor.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.pinot.spi.plugin;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Collection;
+import java.util.HashSet;
+
+
+class PluginsDirectoryVisitor extends SimpleFileVisitor<Path> {
```

Comment:
nit: I don't like having these individual classes that are designed to resolve a problem that is specific to some other class. It is not important, but I would prefer to refactor this class as a inner class of PluginManager.